client/http/http.c

```diff
@@ -73,31 +57,32 @@ void sslWrite(connection *c, char *text);
 
 // Establish a regular tcp connection
 connection * tcpConnect(const char* h, int port)
-{	
-	int error, sock = 0 ;
+{
+	int sock = 0 ;
 	struct sockaddr_in server;
 	if(!strcmp(h, "any")) {
 		server.sin_addr.s_addr = htonl(INADDR_ANY);
 	} else if(!inet_aton(h, &server.sin_addr)) {
 		struct hostent *host = gethostbyname(h);
-		sock = socket(AF_INET, SOCK_STREAM, 0);
-		if(sock == -1) {
-			xdag_err("Create sock error, %s", strerror(sock));
+		server.sin_addr = *((struct in_addr *) host->h_addr);
+	}
+	
+	sock = socket(AF_INET, SOCK_STREAM, 0);
```

Comment:
Not a big deal, but it makes sense to combine declaration and assignment of variable sock.